src/mgmt/srv_pool.c

```diff
@@ -306,27 +301,37 @@ ds_mgmt_destroy_pool(uuid_t pool_uuid, d_rank_list_t *svc_ranks,
 	rc = ds_pool_svc_check_evict(pool_uuid, svc_ranks, NULL, 0, true,
 				     force);
 	if (rc != 0) {
-		D_ERROR("Failed to check/evict pool handles "DF_UUID" rc: %d\n",
-			DP_UUID(pool_uuid), rc);
+		D_ERROR("Failed to check/evict pool handles " DF_UUID ", "
+			DF_RC "\n",  DP_UUID(pool_uuid), DP_RC(rc));
 		goto out;
 	}
 
-	rc = ds_pool_svc_destroy(pool_uuid);
-	if (rc != 0) {
-		D_ERROR("Failed to destroy pool service "DF_UUID": "DF_RC"\n",
-			DP_UUID(pool_uuid), DP_RC(rc));
+	/* Ask PS for list of storage ranks (tgt corpc destinations) */
+	rc = ds_pool_svc_ranks_get(pool_uuid, svc_ranks, &ranks);
```

Comment:
I understand this has come up as an idea in the past. I have not spent a lot of time thinking about how it would  work and what other challenges it would add. I'm sorry at the moment I can't remember some of the detailed points. Certainly not feasible for me to implement it in this patch.